src/db/transactions/queueTx.ts

```diff
@@ -41,6 +42,10 @@ export const queueTx = async ({
     value: BigNumber.from(await tx.getValue()).toHexString(),
   };
 
+  if (shouldSimulate) {
```

Comment:
we also need to need to check `!deployedContractAddress` as we already simulate to get the address when deploying contract.

it was there earlier, [link](https://github.com/thirdweb-dev/engine/pull/402/files#diff-431cbd487d80223caf92d885eb093aaca1a2b370499fdac32064a69d6918edfdL64)